src/pools/D3MAaveDaiPool.sol

```diff
@@ -0,0 +1,212 @@
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        if (what == "king") king = data;
+        else revert("D3MAaveDaiPool/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    function hope(address hub) external override auth{
+        CanLike(D3mHubLike(hub).vat()).hope(hub);
+    }
+
+    function nope(address hub) external override auth{
+        CanLike(D3mHubLike(hub).vat()).nope(hub);
+    }
+
+    // Deposits Dai to Aave in exchange for adai which gets sent to the msg.sender
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#deposit
+    function deposit(uint256 amt) external override auth {
+        uint256 scaledPrev = adai.scaledBalanceOf(address(this));
+
+        pool.deposit(address(asset), amt, address(this), 0);
+
+        // Verify the correct amount of adai shows up
+        uint256 interestIndex = pool.getReserveNormalizedIncome(address(asset));
+        uint256 scaledAmount = _rdiv(amt, interestIndex);
+        require(adai.scaledBalanceOf(address(this)) >= _add(scaledPrev, scaledAmount), "D3MAaveDaiPool/incorrect-share-credit");
+    }
+
+    // Withdraws Dai from Aave in exchange for adai
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
+    function withdraw(uint256 amt) external override auth {
+        pool.withdraw(address(asset), amt, address(msg.sender));
+    }
+
+    function transfer(address dst, uint256 amt) external override auth returns (bool) {
+        return adai.transfer(dst, amt);
+    }
+
+    function transferAll(address dst) external override auth returns (bool) {
+        return adai.transfer(dst, adai.balanceOf(address(this)));
+    }
+
+    function accrueIfNeeded() external override {}
+
+    // --- Balance of the underlying asset (Dai)
+    function assetBalance() public view override returns (uint256) {
+        return adai.balanceOf(address(this));
+    }
+
+    function maxDeposit() external view override returns (uint256) {
+        return type(uint256).max;
+    }
+
+    function maxWithdraw() external view override returns (uint256) {
+        return _min(asset.balanceOf(address(adai)), assetBalance());
+    }
+
+    function recoverTokens(address token, address dst, uint256 amt) external override auth returns (bool) {
+        return TokenLike(token).transfer(dst, amt);
+    }
+
+    function active() external view override returns (bool) {
+        return true;
+    }
+
+    // --- Collect any rewards ---
+    function collect() external returns (uint256 amt) {
+        require(king != address(0), "D3MAaveDaiPool/king-not-set");
+
+        address[] memory assets = new address[](1);
+        assets[0] = address(adai);
+
+        amt = rewardsClaimer.claimRewards(assets, type(uint256).max, king);
```

Comment:
Consider getting the rewards claimer from `adai` each time here:
https://github.com/aave/protocol-v2/blob/master/contracts/protocol/tokenization/AToken.sol#L297

As this function is rarely called that should be fine gas-wise and probably simplify things (not needing to worry if they change the rewardsClaimer address).

Reply:
done 43ce23f